server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -59,13 +59,28 @@
 import static java.util.Map.entry;
 import static org.elasticsearch.core.Strings.format;
 
+/**
+ * Manages all the Java thread pools we create. {@link Names} contains a list of the thread pools, but plugins can dynamically add more
+ * thread pools to instantiate.
+ */
 public class ThreadPool implements ReportingService<ThreadPoolInfo>, Scheduler {
 
     private static final Logger logger = LogManager.getLogger(ThreadPool.class);
 
     public static class Names {
+        /**
+         * Often used for quick operations not worth the costs of switching to another thread. Also used to avoid the cost of re-queuing
+         * work while leveraging the interface for ease of coding. This should not be used on Netty transport threads if the task may not
+         * always complete quickly.
+         */
         public static final String SAME = "same";
+        /** All the tasks that do not fit into the other categories. Try to avoid this if possible. */
```

Comment:
Again, we need some explanation of what exactly this value does

Reply:
Hmm. All the Names are names of different thread pools that we instantiate in the ThreadPool constructor. So the value doesn't do anything, exactly, rather it's the name of a thread pool. Would adding a comment on Names help to understand that? Or is there something else you don't understand about GENERIC in particular?

Reply:
I mean, an explanation of what threadpool this refers to, maybe a bit about its default settings, and what sort of tasks should be run on it

Reply:
I don't think discussing the default settings is relevant here. Those are set in the ThreadPool constructor. I added a reference to the constructor in the Names comment. 

As to what type of tasks, I don't think there is a strong type to describe here. I believe whatever doesn't reasonably fall under one of the non-generic thread pools is scheduled on this generic one, and that's what I've tried to convey in the comment.

Reply:
Fair enough